libs/crosstarget-utils/src/native/regex.rs

```diff
@@ -0,0 +1,44 @@
+use regex::{Regex as NativeRegex, RegexBuilder};
+
+use crate::common::{RegExpError, RegExpFlags};
+
+pub struct RegExp {
+    inner: NativeRegex,
+}
+
+impl RegExp {
```

Comment:
Should we have a common trait that describes the interface so that it's easier to keep the implementations in sync?

Reply:
You also wouldn't need to duplicate the doc comments then.

Reply:
> Should we have a common trait that describes the interface so that it's easier to keep the implementations in sync?

That would be the first time in the `crosstarget-utils` crate.
The original idea was that the implementation usage in code areas compiled both to native targets and to `wasm32-*` was enough to guarantee the same unified type signature.

We're free to do that if we choose, but not in this PR imho

Reply:
I've changed my mind, a common trait could be useful.